scalardb/src/scalardb/core.clj

```diff
@@ -115,18 +115,20 @@
 
 (defn- prepare-service!
   [test mode]
-  (condp = mode
-    :storage (close-storage! test)
-    :transaction (close-transaction! test)
-    :2pc (close-2pc! test))
   (info "reconnecting to the cluster")
   (loop [tries RETRIES]
     (when (< tries RETRIES)
       (c/exponential-backoff (- RETRIES tries)))
     (if-not (pos? tries)
       (warn "Failed to connect to the cluster")
       (if-let [instance (create-service-instance test mode)]
-        (reset! (mode test) instance)
+        (do
+          (condp = mode
+            :storage (close-storage! test)
+            :transaction (close-transaction! test)
+            :2pc (close-2pc! test))
+          (reset! (mode test) instance)
+          (info "reconnected to the cluster"))
```

Comment:
Just a comment.
This function returns `nil`. It returned `instance` before. It was not good because the `instance` should be used via `test`. (There is no caller using the returned `instance` for now.)
The change looks better.